android/gltfio-android/src/main/java/com/google/android/filament/gltfio/Animator.java

```diff
@@ -37,11 +37,30 @@
  */
 public class Animator {
     private long mNativeObject;
+    private Boolean mIsOwner = false;
 
     Animator(long nativeObject) {
         mNativeObject = nativeObject;
     }
 
+    public Animator(FilamentAsset asset, FilamentInstance instance) {
+        mNativeObject = nCreateAnimatorFromAssetAndInstance(asset.getNativeObject(), instance.getNativeObject());
+        mIsOwner = true;
+    }
+    
+    @Override
+    public void finalize() {
+        try {
+            super.finalize();
+        } catch (Throwable t) { // Ignore
+        } finally {
+            if (mIsOwner) {
+                nDestroyAnimator(mNativeObject);
+                mNativeObject = 0;
+            }
+        }
+    }
```

Comment:
I don't like this -- I think we can't rely on finalizers from being called. I'm not sure what the correct pattern would be here. @romainguy what do you think?

Reply:
Finalizers are unfortunately not reliable (and at best take a lot of time to be invoked). We did use them in a couple of places but it's better to have a manual destroy().